sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/EventGridExtensionConfigProvider.cs

```diff
@@ -174,6 +184,50 @@ private async Task<HttpResponseMessage> ProcessEventsAsync(JArray events, string
             return new HttpResponseMessage(HttpStatusCode.Accepted);
         }
 
+        private async Task<FunctionResult> ExecuteWithTracingAsync(string functionName, TriggeredFunctionData triggerData)
+        {
+            using DiagnosticScope scope = _diagnosticScopeFactory.CreateScope(DiagnosticScopeName, DiagnosticScope.ActivityKind.Consumer);
+            if (scope.IsEnabled)
+            {
+                if (triggerData.TriggerValue is JArray evntArray)
+                {
+                    foreach (JToken eventToken in evntArray)
+                    {
+                        AddLinkIfEventHasContext(scope, eventToken);
+                    }
+                }
+                else if (triggerData.TriggerValue is JToken eventToken)
+                {
+                    AddLinkIfEventHasContext(scope, eventToken);
+                }
+            }
+
+            scope.Start();
```

Comment:
Should we start the DiagnosticScope only if it's enabled?

Reply:
This is already handled within DiagnosticScope, so we use the pattern here to avoid adding the same boilerplate code everywhere.